src/js/components/TextInput/TextInput.js

```diff
@@ -267,7 +267,12 @@ class TextInput extends Component {
     }
   };
 
-  onDropClose = () => {
+  onDropClose = event => {
```

Comment:
i think we could just always listen for the `onEsc` regardless of having suggestions or not